.github/workflows/iroha2-dev.yml

```diff
@@ -12,84 +13,78 @@ env:
+          driver: docker
+
+      - name: Build iroha2:base image
+        uses: docker/build-push-action@v3
         with:
+          load: true
+          tags: iroha2:base
           context: .
-          push: true
-          tags: hyperledger/iroha2:dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-      - name: Build and push docker image (iroha_client_cli)
-        uses: docker/build-push-action@v2
+          file: Dockerfile.base
+
+      - name: Build iroha2:build docker image
+        uses: docker/build-push-action@v3
         with:
+          load: true
+          tags: iroha2:build
           context: .
-          push: true
-          tags: hyperledger/iroha2:client-cli-dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-            BIN=iroha_client_cli
-      - name: Build and push docker image (kagami)
-        uses: docker/build-push-action@v2
+          file: Dockerfile.build
+
+      - name: Build iroha2:dev docker image
+        run: |
+          docker build -f Dockerfile -t hyperledger/iroha2:dev .
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
         with:
-          context: .
-          push: true
-          tags: hyperledger/iroha2:kagami-dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-            BIN=kagami
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Push iroha2:dev docker image
+        run: |
+          docker push hyperledger/iroha2:dev
 
       - name: Build and push docker image (load-rs:dev)
         run: |
           sleep 10s
           echo "wait to other workflow"
-      - uses: convictional/trigger-workflow-and-wait@v1.6.0
+      - uses: convictional/trigger-workflow-and-wait@v1.6.2
         with:
           owner: soramitsu
           repo: iroha2-longevity-load-rs
           github_token: ${{ secrets.G_ACCESS_TOKEN }}
           workflow_file_name: load-rs-push-from-dev.yaml
           ref: dev
 
-  build-ci-image:
+  upload_coverage:
+    if: github.event.pull_request.merged == true
     runs-on: ubuntu-latest
     container:
       image: 7272721/i2-ci:latest
     steps:
       - uses: actions/checkout@v2
-      - uses: Swatinem/rust-cache@v1
-      - name: Login to DockerHub
-        uses: docker/login-action@v1
+      - name: Download codecov report artifact
+        uses: dawidd6/action-download-artifact@v2.21.1
         with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Build & push Docker container
-        uses: docker/build-push-action@v2
+          workflow: iroha2-dev-pr.yml
+      - name: Upload to codecov
+        uses: codecov/codecov-action@v3
         with:
-          file: Dockerfile.ci
-          tags: hyperledger/iroha2-ci:latest
-          push: true
+          file: lcov.info
```

Comment:
Now that you removed the renaming step, I don't think this corresponds to `lcov-${{ github.sha }}.info` uploaded by the PR workflow.

Regarding the download failure mentioned in https://github.com/hyperledger/iroha/pull/2433#discussion_r921399893,
have you tried unifying the artifact name to either `lcov-${{ github.sha }}.info` or `lcov-${{ github.event.pull_request.head.sha }}.info` between the upload on PR and the download on merge? 
On my second thought, since both workflows run on `pull_request` event, there was no reason to use different expressions

Reply:
Commit's SHA is changed on PR push and PR merge events. However, trying `{{ github.event.pull_request.head.sha }}` I got this on my fork
![image](https://user-images.githubusercontent.com/66615487/179273948-f9eb6269-924e-4c19-aeb2-122f0d157fab.png)

Reply:
So neither of these expressions can work for passing artifacts?

Reply:
I guess `{{ github.event.pull_request.head.sha }}` could work to pass artifacts for [PR last push and PR merge commits](https://codecov.io/gh/BAStos525/soramitsu-iroha/commit/3edfd1b3969f0223adb0cecf05caf26d5a65a2e4/build).

Reply:
Ah, aren't the `267..` numbers the run ids? No wonder they are different

Reply:
I think `267...` could be different. The main thing here that `commits.sha` for codecov coverage are matched.

Reply:
Yeah, exactly. I think `github.sha` is more appropriate since it points the merge commit:

> Note that `GITHUB_SHA` for this event is the last merge commit of the pull request merge branch. If you want to get the commit ID for the last commit to the head branch of the pull request, use `github.event.pull_request.head.sha` instead.

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

Reply:
Nope, if "the pull request merge branch" means the base branch (`iroha2-dev`), its last merge commit will be useless. In that case, `${{ github.event.pull_request.head.sha }}` will be the only option

Reply:
From practical and experimental side I can say that only `${{ github.event.pull_request.head.sha }}` worked for me. `${{ github.sha }}` showed different value on PR merge action to base branch `iroha2-dev` from PR last push action.

Reply:
Then I think we can resolve this conversation. I'll approve after the [format check](https://github.com/hyperledger/iroha/runs/7362025968?check_suite_focus=true)

Reply:
Please fetch [this commit to fix the format](https://github.com/s8sato/iroha/tree/adda15f37aca01fb444fbd315202f182471c9f4e) if you like

Reply:
> Then I think we can resolve this conversation. I'll approve after passing the [format check](https://github.com/hyperledger/iroha/runs/7362025968?check_suite_focus=true)

Ok, I'll proceed with [remained tasks](https://github.com/hyperledger/iroha/issues/2461) then